sdk/keyvault/azure-security-keyvault-common/inc/azure/keyvault/common/keyvault_pipeline.hpp

```diff
@@ -0,0 +1,79 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#pragma once
+
+#include <azure/core/context.hpp>
+#include <azure/core/http/http.hpp>
+#include <azure/core/http/pipeline.hpp>
+#include <azure/core/response.hpp>
+
+#include <functional>
+#include <memory>
+#include <string>
+#include <vector>
+
+namespace Azure { namespace Security { namespace KeyVault { namespace Common { namespace Internal {
+
+  /**
+   * @brief The HTTP pipeline used by KeyVault clients.
+   */
+  class KeyVaultPipeline {
```

Comment:
Saw this approach in .NET implementation and I liked it!
This class is a wrapper for the client's pipeline and it becomes super useful for all the code duplication that it saves.

The client ctor is still creating the policies, but then it uses this container for holding the pipeline, and the same info that would be directly in the client otherwise.

Reply:
Talked to Jeff about this offline. He approved the KeyVault Pipeline